kani-driver/src/assess/metadata.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Assess metadata. This format is shared between 'assess' and 'assess scan'.
+//! Assess produces this for one workspace, scan for several.
+//! It is not a stable file format: it is meant for assess to directly communicate
+//! from assess subprocesses to a parent scan process.
+//! We can build other tools that make use of it, but they need to built for or distributed
+//! with the specific version of Kani.
+
+use std::fs::File;
+use std::io::BufWriter;
+use std::path::Path;
+
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+
+use super::table_builder::TableBuilder;
+use super::table_failure_reasons::FailureReasonsTableRow;
+use super::table_promising_tests::PromisingTestsTableRow;
+use super::table_unsupported_features::UnsupportedFeaturesTableRow;
+use super::AssessArgs;
+
+/// The structure of `.kani-assess-metadata.json` files, which are emitted for each crate.
+/// This is not a stable interface.
+#[derive(Serialize, Deserialize)]
+pub struct AssessMetadata {
+    /// Report on the presence of `codegen_unimplemented` in the analyzed packages
+    pub unsupported_features: TableBuilder<UnsupportedFeaturesTableRow>,
+    /// Report of the reasons why tests could not be analyzed by Kani
+    pub failure_reasons: TableBuilder<FailureReasonsTableRow>,
+    /// Report on the tests that Kani can successfully analyze
+    pub promising_tests: TableBuilder<PromisingTestsTableRow>,
+}
+
+pub(crate) fn write_metadata(args: &AssessArgs, build: AssessMetadata) -> Result<()> {
+    if let Some(path) = &args.emit_metadata {
+        let out_file = File::create(&path)?;
+        let writer = BufWriter::new(out_file);
+        // use pretty for now to keep things readable and debuggable, but this should change eventually
+        serde_json::to_writer_pretty(writer, &build)?;
+    }
+    Ok(())
+}
+
+pub(crate) fn write_partial_metadata(
```

Comment:
Does "partial" refer to the metadata from the codegen stage only? Can you clarify this with a comment?